### PR TITLE
Boss fight

### PR DESCRIPTION
**Pull request by Westlanderz**
This PR adds a boss fight. It only does not add sprites/animations to the boss and not all of the stages of that boss fight.

### CODE REVIEW

File: Assets/Scripts/UI/GlobalHealthBar.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		}
+
+		// Set up the total Mercenary Bar using the given values
+		public void Setup(float fillValueMax_, int barAmount) {
+			this.fillValueMax_ = fillValueMax_;
+			fillValue_ = 0f;
+			fillValuePerBar_ = fillValueMax_ / barAmount;
+
+			barList_ = new List<Bar>();
+			for (int i = 0; i < barAmount; i++) {
```
Comment (Dennisbonke) [inline Assets/Scripts/UI/GlobalHealthBar.cs:34]: Code style violation, no space between `for` and `(`.

File: Assets/Scripts/UI/GlobalHealthBar.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+		private void OnHealthChanged(object sender, float health) {
+			SetFillValue(health);
+		}
+
+		// Set the fill value for the entire Mercenary Bar, calculate the fillAmount for each inner bar
+		private void SetFillValue(float fillValue_) {
+			this.fillValue_ = fillValue_;
+
+			for (int i = 0; i < barList_.Count; i++) {
```
Comment (Dennisbonke) [inline Assets/Scripts/UI/GlobalHealthBar.cs:64]: Code style violation, no space between `for` and `(`.

File: Assets/Scripts/UI/GlobalHealthBar.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+				Bar bar = barList_[i];
+				float barValueMin = i * fillValuePerBar_;
+				float barValueMax = (i + 1) * fillValuePerBar_;
+				if (fillValue_ <= barValueMin) {
```
Comment (Dennisbonke) [inline Assets/Scripts/UI/GlobalHealthBar.cs:68]: Code style violation, no space between `if` and `(`.

File: Assets/Scripts/UI/GlobalHealthBar.cs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+					// Fill value doesnt reach this bar
+					bar.SetFillAmount(0f);
+				} else {
+					if (fillValue_ >= barValueMax) {
```
Comment (Dennisbonke) [inline Assets/Scripts/UI/GlobalHealthBar.cs:72]: Code style violation, no space between `if` and `(`.

File: Assets/Scripts/UI/GlobalHealthBar.cs [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+			public Bar(RectTransform barRectTransform) {
+				this.barRectTransform = barRectTransform;
+				barFilledImage = barRectTransform.Find("fill").GetComponent<Image>();
+
+				SetFillAmount(0f);
+			}
+
+			public void SetFillAmount(float fillAmount) {
+				if (fillAmount >= 1f) {
```
Comment (Dennisbonke) [inline Assets/Scripts/UI/GlobalHealthBar.cs:117]: Code style violation, no space between `if` and `(`.

File: Assets/Scripts/UI/GlobalHealthBar.cs [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+					if (barFilledImage.fillAmount < 1f) {
```
Comment (Dennisbonke) [inline Assets/Scripts/UI/GlobalHealthBar.cs:118]: Code style violation, no space between `if` and `(`.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
Some small code style violations. Looks good otherwise.

**Review by drfdaniel — DISMISSED** [review]
LGTM

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**